### PR TITLE
Update test coverage and documentation for usage of environment variables as OrtConfiguration parameters

### PR DESCRIPTION
**Pull request by KorSin**
As described in the  [hoplite ReadMe](https://github.com/sksamuel/hoplite#property-sources), environment variables are loaded by default. Furthermore, environment variables have the highest priority which is hard-coded at [at this place](https://github.com/sksamuel/hoplite/blob/7b43bf7eddbbb2f16fd6eeb2287e1035bce4db07/hoplite-core/src/main/kotlin/com/sksamuel/hoplite/PropertySource.kt#L52-L57) and I could not find a simple workaround. The documentation has been updated accordingly.

### CODE REVIEW

File: model/src/test/kotlin/config/OrtConfigurationTest.kt [hunk 1]
```diff
@@ -108,19 +108,25 @@ class OrtConfigurationTest : WordSpec({
...
             )
 
-            val config = OrtConfiguration.load(
-                args = mapOf(
-                    "ort.scanner.storages.postgresStorage.schema" to "argsSchema",
-                    "other.property" to "someValue"
-                ),
-                configFile = configFile
-            )
+            val environmentalVariables = mapOf("ort.scanner.storages.postgresStorage.password" to "envPassword")
```
Comment (sschuberth) [inline model/src/test/kotlin/config/OrtConfigurationTest.kt:111]: Commit message nits: Please write "a test" instead of just "test" in the commit message title and body; in general simple write complete sentence like in a printed book.

Also, mentioning "the default setup" is a bit misleading here as it sounds as if Hoplite was configurable in this regard, but my understanding from you explanations is that Hoplite hard-codes environment variables to take precedence. Finally, "hoplite" should be capitalized as "Hoplite".

So bottom line, I'd propose the commit message to be:

```
OrtConfigurationTest: Add a test for reading environment variables

Add a test that verifies that the Hoplite library supports environment
variables by default and that they take precedence over (command
line) arguments.
```

File: model/src/test/kotlin/config/OrtConfigurationTest.kt [hunk 2]
```diff
@@ -192,6 +198,30 @@ class OrtConfigurationTest : WordSpec({
...
+
+        "support environmental variables" {
+            val user = "user"
+            val password = "password"
+            val url = "url"
+            val schema = "schema"
+            val env = mapOf("ort.scanner.storages.postgresStorage.username" to user,
+                "ort.scanner.storages.postgresStorage.url" to url,
+                "ort__scanner__storages__postgresStorage__schema" to schema,
+                "ort__scanner__storages__postgresStorage__password" to password)
```
Comment (sschuberth) [inline model/src/test/kotlin/config/OrtConfigurationTest.kt:210]: Please format this like

            val env = mapOf(
                "ort.scanner.storages.postgresStorage.username" to user,
                "ort.scanner.storages.postgresStorage.url" to url,
                "ort__scanner__storages__postgresStorage__schema" to schema,
                "ort__scanner__storages__postgresStorage__password" to password
            )

File: model/src/test/kotlin/config/OrtConfigurationTest.kt (hunk 1 above)
Comment (sschuberth) [reply model/src/test/kotlin/config/OrtConfigurationTest.kt:111]: Below, you call `environmentalVariables` just `env`, so I think you can and should do the same here.

File: README.md [hunk 3]
```diff
@@ -189,10 +189,17 @@ of the main ORT configuration file. It consists of sections related to different
 of these sections and the properties they can contain is described together with the corresponding sub components.
 
 While the file is rather static, there are means to override configuration options for a specific run of ORT or to
-customize the configuration to a specific environment. The following options are supported:
-
-* You can override the values of properties on the command line using the `-P` option. The option expects a
-  key-value pair. The key must define the full path to the property to be overridden, e.g.
+customize the configuration to a specific environment. The following options are supported whereas the priority is in the
```
Comment (sschuberth) [inline README.md:192]: Commit message nits: Drop ".md" from "README.md". The "Closes #3242." line should come between body and signed-off footer.

Comment (sschuberth) [reply README.md:192]: How about "The following options are supported, in order of precedence:"

File: README.md [hunk 4]
```diff
@@ -189,10 +189,17 @@ of the main ORT configuration file. It consists of sections related to different
...
+same order as presented:
+
+* Properties can be overridden with environment variables by labeling a variable with the full property path.
```
Comment (sschuberth) [inline README.md:195]: I think it now becomes strange to use the term "overridden" here, as this is the highest priority which overrides nothing that was declared before. Instead of using "overridden" I just say "Properties can be defined ...". Similar in more places below.

Also, what do you mean by "labeling" a variable? I believe you just mean "naming". So how about:

"Properties can be defined via environment variables by using the full property path as the variable name."

File: README.md [hunk 5]
```diff
@@ -189,10 +189,17 @@ of the main ORT configuration file. It consists of sections related to different
...
+  For instance, one can override the postgres schema by setting
```
Comment (sschuberth) [inline README.md:196]: "Postgres"

File: README.md [hunk 6]
```diff
@@ -189,10 +189,17 @@ of the main ORT configuration file. It consists of sections related to different
...
-
-* You can override the values of properties on the command line using the `-P` option. The option expects a
-  key-value pair. The key must define the full path to the property to be overridden, e.g.
+customize the configuration to a specific environment. The following options are supported whereas the priority is in the
+same order as presented:
+
+* Properties can be overridden with environment variables by labeling a variable with the full property path.
+  For instance, one can override the postgres schema by setting 
+  `ort.scanner.storages.postgresStorage.schema=test_schema`. The variable's name is case sensitive.
+  Some programs like bash do not support the "dots" in variable names. For this case, the "dots" can be
```
Comment (sschuberth) [inline README.md:198]: "Bash", no quoting of *dots* or *underscores*

**Review by sschuberth — CHANGES_REQUESTED** [review]

**Review by sschuberth — APPROVED** [review]
